sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestHelper.scala

```diff
@@ -71,6 +76,30 @@ trait SQLQueryTestHelper {
     if (isSorted(df.queryExecution.analyzed)) (schema, answer) else (schema, answer.sorted)
   }
 
+  private def toJson(e: SparkThrowable): String = {
```

Comment:
I'm wondering whether we have messed up the order of the PRs here.
In my mental model I had us implement machineReadable Error message and then simply "flip the config" to produce the golden files in that format. What we have here now is a hand crafted JSON. Is that temporary?

Reply:
Generating the JSON file is a small piece of code. Don't see any problem to remove/move it in the near future. And introduce the config, and then check it on the already existing golden files.

> What we have here now is a hand crafted JSON.

How else would you generate JSON from `SparkThrowable`, I wonder.

> Is that temporary?

Not final.